### PR TITLE
refactor: strict mode, more helpful return types

### PR DESCRIPTION
**Pull request by kanadgupta**
## 🧰 Changes

- Enabled strict mode in `oas-normalize`, `oas-to-har`, and `oas-to-snippet`
- Added more specific types across the board in `oas-normalize`. It's always bugged me that `.validate()` returns a `Promise<any>`.

## 🧬 QA & Testing

Mostly stricter types, no real functional changes (except maybe slightly better error handling in certain cases).

### CODE REVIEW

File: packages/oas-normalize/src/index.ts [hunk 1]
```diff
@@ -84,8 +84,10 @@ export default class OASNormalize {
   /**
    * @private
    */
-  static convertPostmanToOpenAPI(schema: any) {
-    return postmanToOpenAPI(JSON.stringify(schema), null, { outputFormat: 'json', replaceVars: true }).then(JSON.parse);
+  static async convertPostmanToOpenAPI(schema: any) {
```
Comment (kanadgupta) [inline packages/oas-normalize/src/index.ts:87]: added `async` here to address this warning:

CleanShot 2024-04-09 at 14 06 59@2x

wish there was a way for `tsc` to actually throw on these things 🧐

File: packages/oas-to-har/src/index.ts [hunk 2]
```diff
@@ -274,7 +278,10 @@ export default function oasToHar(
     postData: {},
     bodySize: 0,
     method: operation.method.toUpperCase(),
-    url: `${oas.url(formData.server.selected, formData.server.variables)}${operation.path}`.replace(/\s/g, '%20'),
+    url: `${oas.url(formData.server.selected, formData.server.variables as ServerVariable)}${operation.path}`.replace(
+      /\s/g,
+      '%20',
```
Comment (erunion) [inline packages/oas-to-har/src/index.ts:283]: these from prettier?

File: packages/oas-to-snippet/src/index.ts [hunk 3]
```diff
@@ -78,13 +78,17 @@ export default function oasToSnippet(
...
+  }
+
   const har = opts.harOverride || generateHar(oas, operation, values, auth);
   const snippet = new HTTPSnippet(har as HarRequest, {
     // We should only expect HAR's generated with `@readme/oas-to-har` to already be encoded.
     harIsAlreadyEncoded: !opts.harOverride,
   });
 
-  let targetOpts = config.httpsnippet.targets[target].opts || {};
+  let targetOpts = config.httpsnippet.targets[target || ''].opts || {};
```
Comment (erunion) [inline packages/oas-to-snippet/src/index.ts:91]: `target` shouldn't be undefined here anymore with the logic you added above.

Comment (kanadgupta) [reply packages/oas-to-snippet/src/index.ts:91]: good catch, will revert

File: packages/oas-to-har/src/index.ts (hunk 2 above)
Comment (kanadgupta) [reply packages/oas-to-har/src/index.ts:283]: yes 🙃

**Review by erunion — APPROVED** [review]

File: packages/oas-to-snippet/src/index.ts (hunk 3 above)
Comment (kanadgupta) [reply packages/oas-to-snippet/src/index.ts]: https://github.com/readmeio/oas/pull/857/commits/9ee31d3473a00219918201e1c17cfc3ff46ee325

**Pull request merged**